lib/puppet/functions/any_resources_of_type.rb

```diff
@@ -0,0 +1,23 @@
+Puppet::Functions.create_function(:any_resources_of_type, Puppet::Functions::InternalFunction) do 
+  dispatch :any_resources_of_type do
+    scope_param
+
+    required_param 'String', :resource_type
+    optional_param 'Hash[Any, Any]', :parameters
+  end
+
+  def any_resources_of_type(scope, resource_type, parameters = nil)
```

Comment:
Seem that this function is quite generic and it could find usages outside of this module. Would be a better place to move this on?(like stdlib pr even Puppet) Maybe not right know but we could think about future usage.

Reply:
@gimmyxd good catch about `capitalize`, I used it prior to adding `String#casecmp` and forgot to remove it

I also thought about having the function in a more general place. I think it would make sense to have it in Puppet, there's also the `defined` function there which serves a similar purpose.

Reply:
I had a concern about calling `to_s` and increase memory usage if there are a lot of resources,  but i think Ruby is smart enough and if you call `to_s` on a String, it will not create a new object as it does when calling it on other Types.